test/qpy_compat/run_tests.sh

```diff
@@ -26,11 +26,30 @@ qiskit_venv="$(pwd -P)/qiskit_venv"
 qiskit_python="$qiskit_venv/bin/python"
 python -m venv "$qiskit_venv"
 # `packaging` is needed for the `get_versions.py` script.
-"$qiskit_venv/bin/pip" install -c "$our_dir/../../constraints.txt" "$our_dir/../.." packaging
+#  symengine is pinned to 0.13 to explicitly test the migration path reusing the venv
+"$qiskit_venv/bin/pip" install -c "$our_dir/../../constraints.txt" "$our_dir/../.." packaging "symengine~=0.13"
```

Comment:
I think it's ok as written - it's good to be explicit that it's specifically something newer than the 0.11 series, which this pin does.

It's just the comment that's a bit misleading, but I strongly suspect we're going to need to re-do this file a little bit as part of making the cache keys for the QPY jobs more reliable - things like generation script requirements will need to be separated off to become part of the cache key - and we'll be able to expand the inter-symengine-version testing once that's fixed as well, because the jobs will run faster.